pyaerocom/io/mscw_ctm/reader.py

```diff
@@ -234,74 +235,91 @@ def _get_namelist_from_folder(self):
         """
         dd = self.data_dir
 
-        dirs = glob.glob(dd + "/*/")
-        namelist = []
-        yrs = []
+        mscwfiles = []
+        for freq in self.FREQ_CODES.keys():
+            mscwfiles.append(self.FILE_FREQ_TEMPLATE.format(freq=freq))
 
-        for d in dirs:
-            if re.match(r".*(?:19|20)\d\d.*", d) is None:
+        folders = []
+        yrs = []
+        for d in os.listdir(dd):
+            if not os.path.isdir(os.path.join(dd, d)):
                 continue
-            yrs.append(d.split("/")[-1])
+            m = re.match(self.YEAR_PATTERN, d)
+            if m is not None:
+                has_mscwfiles = False
+                for f in mscwfiles:
+                    if os.path.exists(os.path.join(dd, d, f)):
+                        has_mscwfiles = True
+                if has_mscwfiles:
+                    yrs.append(int(m.group(1)))
+                    folders.append(os.path.join(dd, d))
+
+        if len(folders) == 0:  # no trends, use folder
+            for f in mscwfiles:
+                if os.path.exists(os.path.join(dd, f)):
+                    folders = [dd]
+            if len(folders) == 0:
+                raise FileNotFoundError(f"no files like {mscwfiles} found in {dd}")
+        else:
+            folders = [d for _, d in sorted(zip(yrs, folders))]
+        return list(set(folders))
 
-            namelist.append(d)
+    @staticmethod
+    @functools.cache
+    def _get_year_from_nc(filename: str) -> int:
+        with xr.open_dataset(filename) as nc:
+            return np.mean(nc["time"][:]).data.astype("datetime64[Y]").astype(int) + 1970
```

Comment:
Isn't EMEP output yearly files? If so, wouldn't it suffice to just take the first element? Or do you think this is safer?

Reply:
The model starts on the 31.12. for spinup. Hourly and daily data can therefore start a day too early. It's safer to take the middle, and doesn't require any extra time to speak of, since xarray anyway reads all coordinate-variables including the time-variable.